frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx

```diff
@@ -0,0 +1,161 @@
+  );
+
+  React.useEffect(() => {
+    const sortThanosRules = _.sortBy(thanosRules, (rule) => alertingRuleStateOrder(rule));
+    dispatch(monitoringSetRules('devRules', sortThanosRules));
+  }, [dispatch, thanosRules]);
+
+  const filteredRules = React.useMemo(() => {
+    if (columnIndex > -1) {
+      setSortBy({ index: columnIndex + 1, direction: sortOrder });
+    }
+    const fRules = getFilteredRows(filters, alertFilters, rules);
+    const { field, orderBy } = listSorts;
+    return applyListSort(fRules, orderBy || listOrderBy, field || _.camelCase(listSortBy));
+  }, [listSorts, columnIndex, sortOrder, filters, rules, listOrderBy, listSortBy]);
+
+  React.useEffect(() => {
+    const tableRows = monitoringAlertRows(filteredRules);
+    _.forEach(tableRows, (tr) => {
+      _.forEach(collapsedRows, (cr) => {
+        if (tr.cells[0].id === cr.cells[0].id) {
+          tr.isOpen = cr.isOpen;
+        }
+      });
+    });
+    setRows(tableRows);
+  }, [collapsedRows, filteredRules]);
+
+  const onCollapse = (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
+    rows[rowKey].isOpen = isOpen;
+    const collapsedRowsIndex = _.findIndex(
+      collapsedRows,
+      (row) => row.cells[0].id === rows[rowKey].cells[0].id,
+    );
+    if (collapsedRowsIndex === -1) {
+      setCollapsedRows([...collapsedRows, rows[rowKey]]);
+    } else if (collapsedRowsIndex > -1 && isOpen) {
+      collapsedRows.splice(collapsedRowsIndex, 1);
+      setCollapsedRows(collapsedRows);
+    }
+    setRows([...rows]);
+  };
+  const handleSort = (_event: React.MouseEvent, index: number, direction: SortByDirection) => {
+    switch (index) {
+      case 1:
+        dispatch(sortList(reduxID, 'name', undefined, direction, 'Name'));
+        break;
+      case 2:
+        dispatch(sortList(reduxID, 'severity', 'alertSeverityOrder', direction, 'Severity'));
+        break;
+      case 3:
+        dispatch(
+          sortList(reduxID, 'alertState', 'alertingRuleStateOrder', direction, 'Alert State'),
+        );
+        break;
+      default:
+        break;
+    }
+    setSortBy({ index, direction });
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>Alerts</title>
+      </Helmet>
+      <div className="odc-monitoring-alerts">
+        <FilterToolbar
+          rowFilters={alertFilters}
+          hideLabelFilter
+          reduxIDs={[reduxID]}
+          textFilter={textFilter}
+          data={rules}
+        />
+        <Table
+          aria-label="Compact expandable table"
+          onCollapse={onCollapse}
+          rows={rows}
+          cells={monitoringAlertColumn}
+          sortBy={sortBy}
+          onSort={handleSort}
+        >
+          <TableHeader />
+          <TableBody />
+        </Table>
+      </div>
+    </>
+  );
+};
+
+const mapStateToProps = (state: RootState): StateProps => {
+  return {
+    rules: state.UI.getIn(['monitoring', 'devRules']),
+    filters: state.k8s.getIn([reduxID, 'filters'])?.toJS(),
```

Comment:
We should avoid calling `toJS` in `mapStateToProps` because it causes a new object to be created each time, which then causes the component to always render even when `filters` hasn't changed.

You can avoid this by calling `filters.toJS()` inside the component instead.